### PR TITLE
Add template to arguments

### PR DESCRIPTION
**Pull request by cheif**
Possible to send an template with the options for the hook, solves #7. Also added some tests for it.

### CODE REVIEW

File: slack.go [hunk 1]
```diff
@@ -8,6 +8,11 @@ import (
 	"net/http"
 )
 
+type ClientImpl interface {
+	SendMessage(*Message) error
```
Comment (bradrydzewski) [inline slack.go]: lets use `type Client interface` for the interface

File: slack.go [hunk 2]
```diff
@@ -8,6 +8,11 @@ import (
...
+	SetUrl(string)
+}
+
 type Client struct {
 	Url string
 }
```
Comment (bradrydzewski) [inline slack.go]: and `type client struct` (lowercase) for the implementation

File: slack.go [hunk 3]
```diff
@@ -69,6 +74,10 @@ func (c *Client) SendMessage(msg *Message) error {
 	return nil
 }
 
+func (c *Client) SetUrl(url string) {
+	c.Url = url
+}
+
 func (m *Message) NewAttachment() *Attachment {
```
Comment (bradrydzewski) [inline slack.go]: and then we can use `NewClient` and return the interface, while passing the url:

```
func NewClient(url string) Client {
    return &client{url}
}
```

**Comment by bradrydzewski** [comment]
after realizing we also needed to implement the new shared template package, I went ahead and made a few of the changes locally and pushed. Sorry for the confusion, we didn't have a standard for notifications templates at the beginning of the week and now we do :)

**Comment by cheif** [comment]
Yup, that sounds like the correct way to do it :)

**Pull request closed without merging**